### PR TITLE
Gb/conv rnn layers

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

File: phygnn/model_interfaces/phygnn_model.py [hunk 1]
```diff
@@ -123,20 +127,19 @@ def train_model(self, features, labels, p, n_batch=16, n_epoch=10,
...
+            (n_observations, n_features). A 3D input should have the shape:
+            (n_observations, n_timesteps, n_features). 4D inputs have not been
+            tested and should be used with caution.
         labels : np.ndarray | pd.DataFrame
-            Known output data in a 2D array or DataFrame. If this is a
-            DataFrame, the index is ignored, the columns are used with
-            self.output_names, and the df is converted into a numpy array for
-            batching and passing to the training algorithm.
+            Known output data in a 2D array or DataFrame.
+            Same dimension rules as x.
```
Comment (MRossol) [inline phygnn/model_interfaces/phygnn_model.py:139]: should this and p be same dimension rules as "features"?

File: phygnn/model_interfaces/phygnn_model.py [hunk 2]
```diff
@@ -146,15 +149,15 @@ def train_model(self, features, labels, p, n_batch=16, n_epoch=10,
             Number of times to iterate on the training data.
         shuffle : bool
             Flag to randomly subset the validation data and batch selection
-            from features and labels.
+            from x and y.
```
Comment (MRossol) [inline phygnn/model_interfaces/phygnn_model.py:152]: I'm confused by the use of features/labels vs x/y

File: phygnn/phygnn.py [hunk 3]
```diff
@@ -149,15 +158,17 @@ def __init__(self, p_fun, loss_weights=(0.5, 0.5),
...
-        self._input_dims = n_features
-        self._output_dims = n_labels
-        self._layers = Layers(n_features, n_labels=n_labels,
-                              hidden_layers=hidden_layers,
-                              input_layer=input_layer,
-                              output_layer=output_layer)
+        self._n_features = n_features
+        self._n_labels = n_labels
+        self._layers = layers_obj
+        if layers_obj is None:
```
Comment (MRossol) [inline phygnn/phygnn.py:167]: Like it! Might be worth adding an:
```
elif not isinstance(layers_obj, Layers):
    raise
```

**Review by MRossol — COMMENTED** [review]
A few questions and one suggestion to check layer_obj type

File: phygnn/model_interfaces/phygnn_model.py (hunk 1 above)
Comment (grantbuster) [reply phygnn/model_interfaces/phygnn_model.py:139]: ah yes. just copied the doc string from phygnn

File: phygnn/model_interfaces/phygnn_model.py (hunk 2 above)
Comment (grantbuster) [reply phygnn/model_interfaces/phygnn_model.py:152]: i just use x and y as kind of shorthand in the phygnn model while the model interfaces prefer the features and labels var names. I just keep copying over doc strings so thats why this is getting messed up. will fix.

File: phygnn/phygnn.py (hunk 3 above)
Comment (grantbuster) [reply phygnn/phygnn.py:167]: good idea!

**Comment by grantbuster** [comment]
@MRossol, this will be a breaking change as the phygnn save object is now different... I think we're all good to release now though? I'm retraining mlclouds anyway.

**Comment by MRossol** [comment]
love breaking changes! go for the release assuming tests pass!

**Pull request merged**